### PR TITLE
test: refine and improve pkg/utils unit test coverage to 82.3%

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What problem does this PR solve? 

part of #998 

### What is changed and how it works?

- add unit test cases
- tiny refine code

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -200,8 +200,8 @@ coverage_fix_cover_mode:
 	sed -i "s/mode: count/mode: atomic/g" $(TEST_DIR)/cov.*.dmctl.*.out
 
 coverage: coverage_fix_cover_mode retool_setup
-	retool do gocovmerge "$(TEST_DIR)"/cov.* | grep -vE ".*.pb.go|.*.__failpoint_binding__.go" > "$(TEST_DIR)/all_cov.out"
-	retool do gocovmerge "$(TEST_DIR)"/cov.unit_test*.out | grep -vE ".*.pb.go|.*.__failpoint_binding__.go" > $(TEST_DIR)/unit_test.out
+	retool do gocovmerge "$(TEST_DIR)"/cov.* | grep -vE ".*.pb.go|.*.__failpoint_binding__.go|.*debug-tools.*|.*portal.*|.*tracer.*|.*loader.*|.*tracing.*|.*ctl.* > "$(TEST_DIR)/all_cov.out"
+	retool do gocovmerge "$(TEST_DIR)"/cov.unit_test*.out | grep -vE ".*.pb.go|.*.__failpoint_binding__.go|.*debug-tools.*|.*portal.*|.*tracer.*|.*loader .*|.*tracing.*|.*ctl.*" > $(TEST_DIR)/unit_test.out
```
Comment (lance6716) [inline Makefile:204]:
```suggestion
	retool do gocovmerge "$(TEST_DIR)"/cov.unit_test*.out | grep -vE ".*.pb.go|.*.__failpoint_binding__.go|.*debug-tools.*|.*portal.*|.*tracer.*|.*loader.*|.*tracing.*|.*ctl.*" > $(TEST_DIR)/unit_test.out
```

**Review by lance6716 — COMMENTED** [review]
rest LGTM

File: pkg/utils/db_test.go [hunk 2]
```diff
@@ -17,11 +17,198 @@ import (
...
+	c.Assert(mock.ExpectationsWereMet(), IsNil)
+
+	// `ANSI_QUOTES`
+	rows = mock.NewRows([]string{"Variable_name", "Value"}).AddRow("sql_mode", "ANSI_QUOTES")
+	mock.ExpectQuery(`SHOW VARIABLES LIKE 'sql_mode'`).WillReturnRows(rows)
+	p, err = GetParser(db)
+	c.Assert(err, IsNil)
+	_, err = p.ParseOneStmt(DDL1, "", "")
+	c.Assert(err, IsNil)
+	_, err = p.ParseOneStmt(DDL2, "", "")
```
Comment (lance6716) [inline pkg/utils/db_test.go:186]: is this test intended to have a SQL like `ALTER TABLE tbl ADD COLUMN "c1" INT` and test it could parse?

**Review by lance6716 — COMMENTED** [review]
rest LGTM

**Comment by lance6716** [comment]
LGTM

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**